e2e/Microsoft.Azure.Devices.E2ETests/MessageE2ETests.NetFramework.cs

```diff
@@ -0,0 +1,235 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.ServiceBus.Messaging;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public partial class MessageE2ETests
+    {
+
+
+        private async Task SendMessageThrottledForHttp()
+        {
+            await sequentialTestSemaphore.WaitAsync();
+
+            Tuple<string, string> deviceInfo = TestUtil.CreateDevice(DevicePrefix, hostName, registryManager);
+
+            EventHubClient eventHubClient;
+            EventHubReceiver eventHubReceiver = CreateEventHubReceiver(deviceInfo.Item1, out eventHubClient);
+
+            var deviceClient = DeviceClient.CreateFromConnectionString(deviceInfo.Item2, Client.TransportType.Http1);
+
+            try
+            {
+                deviceClient.OperationTimeoutInMilliseconds = (uint)TestUtil.ShortRetryInMilliSec;
+                await deviceClient.OpenAsync();
+
+                string payload, p1Value;
+                Client.Message testMessage = ComposeD2CTestMessage(out payload, out p1Value);
+                await deviceClient.SendEventAsync(testMessage);
+
+                bool isReceived = false;
+                Stopwatch sw = new Stopwatch();
+                sw.Start();
+                while (!isReceived && sw.Elapsed.Minutes < 1)
+                {
+                    var events = await eventHubReceiver.ReceiveAsync(int.MaxValue, TimeSpan.FromSeconds(5));
+                    isReceived = VerifyTestMessage(events, deviceInfo.Item1, payload, p1Value);
+                }
+                sw.Stop();
+
+                // Implementation of error injection of throttling on http is that it will throttle the
+                // fault injection message itself only.  The duration of fault has no effect on http throttle.
+                // Client is supposed to retry sending the throttling fault message until operation timeout.
+                await deviceClient.SendEventAsync(TestUtil.ComposeErrorInjectionProperties(TestUtil.FaultType_Throttle,
+                    TestUtil.FaultCloseReason_Boom, TestUtil.DefaultDelayInSec, TestUtil.DefaultDurationInSec));
+            }
+            finally
+            {
+                await deviceClient.CloseAsync();
+                await eventHubReceiver.CloseAsync();
+                sequentialTestSemaphore.Release(1);
+            }
+        }
+
+        private async Task SendMessageRecovery(Client.TransportType transport,
```

Comment:
Please make the PAL layer clear: you can delimit it with a #region or mark the PAL API as internal, etc. It would be nice if the common interface is somewhere in the top part of the file.